addons/website_sale_autocomplete/controllers/main.py

```diff
@@ -0,0 +1,151 @@
+        We try to guess the number from the user's input to avoid losing the info.
+        """
+        # Remove other parts from address to make better guesses
+        guess = source_input.replace(standard_address.get('zip', ''), '') \
+            .replace(standard_address.get('street', ''), '') \
+            .replace(standard_address.get('city', ''), '')
+        guess = guess.split(',')[0].strip()
+        return guess
+
+    def _perform_place_search(self, partial_address, session_uuid=None, language_code=None, country_code=None):
+        params = {
+            'key': self._get_key(),
+            'fields': 'formatted_address,name',
+            'inputtype': 'textquery',
+            'types': 'address',
+            'input': partial_address
+        }
+        if country_code:
+            params['components'] = f'country:{country_code}'
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        try:
+            results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/autocomplete/json', params=params, timeout=TIMEOUT).json()
+        except TimeoutError:
+            return []
+        results = results['predictions']
+
+        # Convert google specific format to standard format.
+        return [{
+            'formatted_address': result['description'],
+            'google_place_id': result['place_id'],
+        } for result in results]
+
+    def _perform_complete_place_search(self, address, google_place_id=None, language_code=None, session_uuid=None):
+        params = {
+            'key': self._get_key(),
+            'place_id': google_place_id,
+            'fields': 'address_component,adr_address'
+        }
+
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        try:
+            results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/details/json', params=params, timeout=TIMEOUT).json()
+        except TimeoutError:
+            return {'address': None}
+
+        html_address = results['result']['adr_address']
+        results = results['result']['address_components']  # Get rid of useless extra data
+
+        # Keep only the first type from the list of types
+        for res in results:
+            res['type'] = res.pop('types')[0]
+
+        # Sort the result by their priority.
+        results.sort(key=lambda r: FIELDS_PRIORITY.index(r['type']) if r['type'] in FIELDS_PRIORITY else 100)
+
+        standard_address = self._translate_google_to_standard(results)
+
+        if 'number' not in standard_address:
+            standard_address['number'] = self._guess_number_from_input(address, standard_address)
+            standard_address['formatted_street_number'] = f'{standard_address["number"]} {standard_address.get("street", "")}'
+        else:
+            formatted_from_html = html2plaintext(html_address.split(',')[0])
```

Comment:
Here I do a bit of HTML parsing, since the Google API only returns the data I want in that format. I have assumed that using `html2plaintext` was safe but correct me if that's not the case

Reply:
html2plaintext is safe if you treat the result as plain text. It's not flagged by teh security check here.